File: 2024/tinhoclk/AIRC.py
n = int(input())
a = list(map(int,input().split())) 
m = int(input())
class D :
    def __init__(self, a, b ) : 
        self.a = a 
        self.b = b 
    def __lt__(self, other) : 
        return self.a<other.a or ( self.a==other.a and self.b<other.b) 
    def __str__(self) : 
        return f"({self.a} {self.b})"
AC = []
for i in range(m) : 
    X,Y = map(int,input().split())  
    AC[i].append(D(X,Y)) 
AC.sort()
mi = int(1e9) 
for i in range(m-1,-1,-1): 
    mi = min(mi,AC[i].b)
    AC[i].b = mi 
 
res = 0 
for i in range(n) : 
    l = 0 
    r = m-1 
    ans = 0
    while l <=r : 
        mid = (l+r)//2
        if AC[mid].a>=a[i] : 
            ans = mid 
            r=mid-1 
        else  : 
            l=mid+1
    res+=AC[ans].b 
print(res) 

File: 2024/tinhoclk/AIRC.cpp
#include<bits/stdc++.h>
using namespace std ; 

const int N = 3e5+5 , oo = 1e9+5 ; 
const long long inf = 1e17+5 , sm = 1e9+7 ; 

int a[N] ;
struct DL
{
    int x, y; 
    bool operator<(const DL&a )const{
        return x<a.x||(x==a.x&&y<a.y); 
    }
} ; 
DL b[N] ;
signed main()
{
    ios_base::sync_with_stdio(0) ; cin.tie(0) ; cout.tie(0) ;
    int n; cin>> n; 
    for(int i=1;i<=n;i++)cin>>a[i] ; 
    int m; cin>> m; 
    for(int i=1;i<=m;i++)
    {
        cin>>b[i].x>>b[i].y;
    }
    sort(b+1,b+m+1) ; 
    int mi = 1e9 ; 
    for(int i=m;i>=1;i--)
    {
        mi = min(mi,b[i].y) ; 
        b[i].y= mi ;
    }
    int res = 0 ; 
    for(int i=1;i<=n;i++)
    {
        int l = 1 ;
        int r = m ; 
        int ans = 0 ; 
        while(l<=r)
        {
            int mid = (l+r)>>1 ;
            if(b[mid].x>=a[i])ans=mid,r=mid-1 ;
            else l=mid+1 ;
        }
        res+=b[ans].y ; 
    }
    cout<<res<<'\n';
}